connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -395,7 +445,35 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
             // *next record*, not the last one consumed.
             offset = record.offset() + 1;
```

Comment:
Shouldn't we update the offset in a synchronized block? Otherwise, the offset can still be updated even we are writing the snapshot?